packages/devextreme/testing/tests/DevExpress.ui.widgets.editors/tagBox.tests.js

```diff
@@ -6822,6 +6822,45 @@ QUnit.module('performance', () => {
         assert.ok($.isFunction(filter), 'filter is function');
     });
 
+    [false, true].forEach(changeAtRuntime => {
+        QUnit.test(`tag labels should be correctly displayed with valueExpr as function and hideSelectedItems enabled ${changeAtRuntime ? 'at runtime' : ''} (T1234032)`, function(assert) {
+            const dataSource = [
+                { id: 1, scheme: 'schema 1', name: 'item1' },
+                { id: 2, scheme: 'schema 2', name: 'item2' },
+                { id: 3, scheme: 'schema 3', name: 'item3' },
+            ];
+
+            const $tagBox = $('#tagBox').dxTagBox({
+                dataSource,
+                valueExpr(x) {
+                    return x && x.name + ' ' + x.scheme;
+                },
+                displayExpr: 'name',
+                hideSelectedItems: !changeAtRuntime,
+                opened: true,
+            });
+
+            const tagBox = $tagBox.dxTagBox('instance');
+            const $list = tagBox._list.$element();
+
+            if(changeAtRuntime) {
```

Comment:
[excess operation]

This condition is excess: if initially property is `true`, setting it to `true` at runtime will do nothing